Sources/GRPC/ConnectionManager.swift

```diff
@@ -582,6 +593,31 @@ internal final class ConnectionManager {
     }
   }
 
+  /// Registers a callback which fires when the current active connection is closed.
+  ///
+  /// If there is a connection, the callback will be invoked with `true` when the connection is
+  /// closed. Otherwise the callback is invoked with `false`.
+  internal func onCurrentConnectionClose(_ onClose: @escaping (Bool) -> Void) {
```

Comment:
This is a somewhat tricky API to use. Is there a reason it doesn't return whether the callback got registered?

Reply:
We need to hop onto the right EL first which makes this slightly awkward. A few options:
 - let the caller provide a promise / return a future
 - expose this via the `sync` view and return whether it was registered (could be in addition to above)
 - use a lock in favour of the EL for synchronising here
 - pass whether the callback was registered into the callback (current api)

I don't think any of these stand out as significantly better choices for an internal API. WDYT?

Reply:
Yeah, that's fair. I dunno man, it's all kinda gross. This works I suppose.